integration/runner/runner.go

```diff
@@ -0,0 +1,184 @@
+// reset (including OS environment variables)
+func (c Cmd) WithCleanEnv() Cmd {
+	c.env = []string{}
+	c.cleanEnv = true
+	return c
+}
+
+// WithTimeout returns a copy of the command with new timeout value
+func (c Cmd) WithTimeout(timeout time.Duration) Cmd {
+	c.timeout = timeout
+	return c
+}
+
+// Start the command and returns underlying session
+func (c Cmd) Start() *gexec.Session {
+	command := exec.Command(c.execPath, c.args...)
+
+	if c.cleanEnv {
+		command.Env = []string{}
+	} else {
+		command.Env = os.Environ()
+	}
+	command.Env = append(command.Env, c.env...)
+
+	fmt.Fprintf(GinkgoWriter, "starting '%s'\n", c.String())
+
+	session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
+	if err != nil {
+		Fail(fmt.Sprintf("error starting process: %v\n", err), 1)
+	}
+
+	return session
+}
+
+// Run the command and wait for it to return
+func (c Cmd) Run() *gexec.Session {
+	session := c.Start()
+	session.Wait(c.timeout)
+	return session
+}
+
+type runCmdMatcher struct {
+	session *gexec.Session
+	args    []string
+
+	failedCmd string
+}
+
+func (m *runCmdMatcher) run(cmd Cmd) bool {
+	m.session = cmd.Run()
+	if m.session.ExitCode() == 0 {
+		return true
+	}
+	m.failedCmd = cmd.String()
+	return false
+}
+
+// RunSuccessfully matches successful excution of a command
+func RunSuccessfully() types.GomegaMatcher {
+	return &runCmdMatcher{}
+}
+
+func (m *runCmdMatcher) Match(actual interface{}) (bool, error) {
+	switch actual.(type) {
+	case Cmd:
+		return m.run(actual.(Cmd)), nil
+	case []Cmd:
+		for _, cmd := range actual.([]Cmd) {
+			if !m.run(cmd) {
+				return false, nil
+			}
+			continue
+		}
+		return true, nil
+	default:
+		return false, fmt.Errorf("not a Cmd or []Cmd")
+	}
+}
+
+func (m *runCmdMatcher) FailureMessage(_ interface{}) string {
+	return fmt.Sprintf("Expected '%s' to succeed, got return code %d", m.failedCmd, m.session.ExitCode())
+}
+
+func (m *runCmdMatcher) NegatedFailureMessage(_ interface{}) string {
+	return fmt.Sprintf("Expected command NOT to succeed")
+}
+
+type runCmdOutputMatcher struct {
+	outputMatcher  types.GomegaMatcher
+	failureMessage string
+	*runCmdMatcher
+}
+
+// RunSuccessfullyWithOutputString matches successful excution of a command and passes the output string
+// to another matcher (the string will include stdout and stderr)
+func RunSuccessfullyWithOutputString(outputMatcher types.GomegaMatcher) types.GomegaMatcher {
+	return &runCmdOutputMatcher{
+		outputMatcher: outputMatcher,
+		runCmdMatcher: &runCmdMatcher{},
+	}
+}
+
+func (m *runCmdOutputMatcher) Match(actual interface{}) (bool, error) {
+	cmd, ok := actual.(Cmd)
+	if !ok {
+		return false, fmt.Errorf("not a Cmd")
+	}
+
+	if !m.runCmdMatcher.run(cmd) {
+		m.failureMessage = m.runCmdMatcher.FailureMessage(nil)
+		return false, nil
+	}
+
+	output := string(m.runCmdMatcher.session.Buffer().Contents())
+
+	if ok, err := m.outputMatcher.Match(output); !ok {
+		m.failureMessage = m.outputMatcher.FailureMessage(output)
```

Comment:
shouldn't we capture the `negatedFailureMessag` here and return it in `runCmdOutputMatcher#NegatedFailureMessage`?

Reply:
Good idea, thanks!